### DISCUSSION TITLE
Turn a Group of Nodes into one node.

### DISCUSSION THREAD

**Discussion by ogabrielluiz** [xyflow/xyflow#3089] [Q&A]
Hello! 

React Flow is awesome. We are loving working with it. 

I was checking the Sub Flows documentation and saw that a group is sort of a node with nodes inside but I'm not sure yet what changes in the interaction when a node has a parent. 

What I wanted to do is basically make a group of nodes have their handles be available through the Parent node. 

Example: Node A has handle 1 and 2. Node B connects to handle 1 of A. I group them and now, Parent Node has handle 2 of A exposed like a proxy of sorts. 

How could I go about implementing that?

**Comment by moklick** [comment]
Hey @ogabrielluiz 

that's great to hear :) 

When you assign `parentNode` to a node it will be positioned relatively to its parent. When you move the parent, the child node moves too. Another feature is that you can set the `extent` option to 'parent'. If this option is set, you can't move a child node out of its parent. 

Regarding your question: 

1. You could create a group node, that updates its handles based on its child nodes. You could access the child nodes like this:
```js 
const childNodes = useStore(s => s.getNodes().filter(node => node.parentId === id));
```

2. Depending on your needs you might not need a subflow at all, but you could render a node that combines its children by just showing their names and their handles. 

Does this help?

**Reply by ogabrielluiz** [reply]
It does help. Thanks!

I'm a maintainer of [LangFlow](https://github.com/logspace-ai/langflow) but I'm more of the Python person on the team, so bear with me.

So, is the Subflow functionality all about positioning? 
What happens to an edge connected to the parent node? Does any of the child nodes know about it? 

Right now, what we are doing is creating a new node that has the entire grouped ReactFlowInstance object in it, and on the backend, I shuffle the node ids to expand the graph into the main graph. This does work in our experiments but I'd rather see if we could find a more official way of doing this, you know?

In our latest discussion we were thinking of creating proxy fields(fields are the input component that may have a handle) in the parent node and that field would have the id of the node it came from. I think this is similar to what you suggested. We'd have to do it this way because we are building predetermined objects in the backend so we have to abide by their structure for now.


Sorry for writing too much.

**Reply by moklick** [reply]
no worries! We saw LangFlow. It's a super interesting project :) 

Do I understand you right, that you are rendering another <ReactFlow /> component inside a node in order to display a subflow?

**Reply by ogabrielluiz** [reply]
When we group the nodes we make them disappear and add a new node that has the grouped nodes in one of it's attributes (props? I'm not familiar with typescript terms).

The flow must be used as if it were the ungrouped nodes but visually it is a Node that acts as an entire flow.

**Reply by moklick** [reply]
That sounds good to me. We do not have helpers for this, since it's very hard to figure out what is needed because data structures and use cases are often so different. If you an idea for some helper functions / hooks that could make your life easier, let us know :)
